src/nupic/os/Regex.cpp

```diff
@@ -46,7 +47,14 @@ namespace nupic
       exactRegExp += re;
       if (re[re.length()-1] != '$') 
         exactRegExp += '$';
-      
+
+#if defined(NTA_OS_WINDOWS)
```

Comment:
Got it, that makes sense. How about we create an issue to swap out pcre when `std::regex` has sufficient support and leave all platforms on pcre for now. That avoids platform special-casing. Does that work?